src/lightning/pytorch/loops/dataloader/evaluation_loop.py

```diff
@@ -88,6 +89,7 @@ def skip(self) -> bool:
         max_batches = self._get_max_batches()
         return sum(max_batches) == 0
 
+    @_no_grad_context
```

Comment:
Does it have to be a decorator? Can't we just keep it as a context manager, moving it directly over the logic in run()?

IMO the decorator approach is more complicated than it has to be. It requires this validation logic even though it is just internal.

Reply:
It doesnt have to be. I just thought it would be more readable like this, as it doesnt add an extra indentation level to the run implementation. Ccing @justusschock for the tiebreaker

Reply:
I agree with @carmocca here that this is more readable.

Reply:
Okay, so we can do the current way. 

I'm still convinced that the ctx manager way is more readable as it has less code and nesting. The extra indentation level is just how it works in Python, so I'm not seeing it as a strong argument.